hazelcast-client/src/test/java/com/hazelcast/client/connection/nio/AdvancedNetworkingClientCommunicationIntegrationTest.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (c) 2008-2019, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.client.connection.nio;
+
+import com.hazelcast.client.HazelcastClient;
+import com.hazelcast.client.config.ClientConfig;
+import com.hazelcast.config.Config;
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.internal.networking.nio.AbstractAdvancedNetworkIntegrationTest;
+import com.hazelcast.test.HazelcastSerialClassRunner;
+import com.hazelcast.test.annotation.SlowTest;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import static org.junit.Assert.fail;
+
+@RunWith(HazelcastSerialClassRunner.class)
+@Category(SlowTest.class)
+public class AdvancedNetworkingClientCommunicationIntegrationTest extends AbstractAdvancedNetworkIntegrationTest {
```

Comment:
Does it make sense to move this test to `AdvancedNetworkClientIntegrationTest`? Maybe not, just saying.

Reply:
@vbekiaris  thanks for review! I would prefer to keep those tests separated. Moving `AdvancedNetworkingClientCommunicationIntegrationTest` to `AdvancedNetworkClientIntegrationTest` means that `AdvancedNetworkClientIntegrationTest` has to extend `AbstractAdvancedNetworkIntegrationTest`. I would like to avoid this dependency since just `testClientConnectionToEndpoints()` is related to that abstract class. It can cause some trouble when `AdvancedNetworkClientIntegrationTest` would need to extend some other class for any reason. Is it ok for you?

Reply:
yup, let's keep them separate